src/nnet3/nnet-simple-component.h

```diff
@@ -309,6 +310,7 @@ class AffineComponent: public UpdatableComponent {
   Component *CollapseWithNext(const FixedAffineComponent &next) const;
```

Comment:
Currently the Vectorize functions do not automatically resize the out vector (*params). Isn't it better to automatically resize it to NumParameters() ?